lib/compat/time.c

```diff
@@ -23,7 +23,7 @@
 
 #include "compat/time.h"
 
-#ifdef __MACH__
+#ifdef __APPLE__
```

Comment:
According to the man page, `clock_gettime()` conforms to
* `SUSv2`
* `POSIX.1-2001`

It also mentions the `glibc` feature check macro
* `_POSIX_C_SOURCE >= 199309L`

Do we have our macros set up correctly?

Which version of OSX does this workaround involve?

They say that Mac OS X is UNIX 03 / POSIX certified since 10.5:
* http://www.opengroup.org/openbrand/register/brand3555.htm
* https://blog.opengroup.org/2012/07/25/apple-registers-mac-os-x-10-8-mountain-lion-to-the-unix-03-standard/
* http://www.opengroup.org/openbrand/register/brand3612.htm

If these hints don't help, as a workaround, I wanted to recommend checking for `clock_gettime()` in `configure` to set a variable, but it looks like we already check it and set `SYSLOG_NG_HAVE_CLOCK_GETTIME` accordingly. Could you double check?